src/coreclr/gc/gc.cpp

```diff
@@ -19241,7 +19191,7 @@ void gc_heap::balance_heaps (alloc_context* acontext)
 
 #ifdef HEAP_BALANCE_INSTRUMENTATION
                         int current_proc_no_before_set_ideal = GCToOSInterface::GetCurrentProcessorNumber ();
-                        if (current_proc_no_before_set_ideal != last_proc_no)
+                        if ((uint16_t)current_proc_no_before_set_ideal != last_proc_no)
```

Comment:
is there a reason why a cast is required here, can it just be defined as uint16?

Reply:
then it needs a cast there because `GCToOSInterface::GetCurrentProcessorNumber` returns a 32-bit number.